scripts/TestGen.hx

```diff
@@ -0,0 +1,380 @@
+        ::stories::
+    }
+}");
+
+	static var storyTmpl = new Template("describe(\"::story::\", {
+        ::testCases::});");
+
+	static var testCaseTmpl = new Template("    it(\"::testCase::\", {
+                ::testCriteria::
+            });
+        ");
+
+	public static function main()
+	{
+		// TODO: Configlet is able to create the directories/files we need:
+		//   ./bin/configlet sync --docs --filepaths --metadata --tests include -uye <slug>
+		// 
+		// We should try to change the code below to use configlet's
+		// cached copy of the problem-specifications repo instead
+		var dataUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/canonical-data.json");
+		var descriptionUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/description.md");
+		var metaUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/metadata.toml");
+		// check if no args are passed
+		if (Sys.args().length < 1)
+		{
+			Sys.println("An exercise is required.");
+			return;
+		}
+
+		exercise = Sys.args()[0];
+
+		// pull in files from github
+		var execute = (template:Template) -> return template.execute({exercise: exercise});
+
+		var dataJson = Http.requestUrl(execute(dataUrl));
+		var descriptionMd = Http.requestUrl(execute(descriptionUrl));
+		var metaToml = Http.requestUrl(execute(metaUrl));
+
+		var exercisesDir = "../exercises/practice/";
+		var dir = '${exercisesDir}${exercise}/';
+
+		// create exercise folder
+		if (!FileSystem.exists(dir))
+		{
+			FileSystem.createDirectory(dir);
+		}
+		// create .meta
+		if (!FileSystem.exists('${dir}.meta'))
+		{
+			FileSystem.createDirectory('${dir}.meta');
+		}
+		// create src folder
+		if (!FileSystem.exists('${dir}src'))
+		{
+			FileSystem.createDirectory('${dir}src');
+		}
+		// create test folder
+		if (!FileSystem.exists('${dir}test'))
+		{
+			FileSystem.createDirectory('${dir}test');
+		}
+
+		// create string for tests.toml; first line is always [canonical-tests]
+		var meta = new Array<String>();
+		meta.push("[canonical-tests]\n\n");
+
+		var problemSpec = Json.parse(dataJson);
+		var testCases = new Map<String, Array<String>>();
+		var stories = new Array<String>();
+		var methods = new Map<String, {order:Int, method:String}>();
+		// populate test cases
+		generateTests(cast(problemSpec.cases, Array<Dynamic>), methods, stories, testCases, meta);
+
+		var renderedMethods = new Array<String>();
+		for (key in methods.keys())
+		{
+			var method = methods.get(key);
+			renderedMethods.insert(method.order, method.method);
+		}
+
+		for (i in 0...stories.length)
+		{
+			stories[i] = storyTmpl.execute(
+				{
+					story: stories[i],
+					testCases: testCases.get(stories[i]).join("")
+				});
+		}
+
+		// create String array for README; add exercise as title
+		var readme = new Array<String>();
```

Comment:
configlet will give us the proper docs synced from problem-specifications so we won't need this section